S7.Net/Types/S7String.cs

```diff
@@ -8,11 +8,18 @@ namespace S7.Net.Types
     /// An S7 String has a preceeding 2 byte header containing its capacity and length
     /// </summary>
     public static class S7String
-    {
-        /// <summary>
-        /// The Encoding used when serializing and deserializing S7String (Encoding.ASCII by default)
-        /// </summary>
-        public static Encoding StringEncoding { get; set; }
+    {
+        private static Encoding stringEncoding;
+
+        /// <summary>
+        /// The Encoding used when serializing and deserializing S7String (Encoding.ASCII by default)
+        /// </summary>
+        /// <exception cref="ArgumentNullException">StringEncoding must not be null</exception>
+        public static Encoding StringEncoding
+        {
+            get => stringEncoding ??= Encoding.ASCII;
+            set => stringEncoding = value ?? throw new ArgumentNullException(nameof(StringEncoding));
+        }
```

Comment:
```suggestion
        private static Encoding stringEncoding = Encoding.ASCII;

        /// <summary>
        /// The Encoding used when serializing and deserializing S7String (Encoding.ASCII by default)
        /// </summary>
        /// <exception cref="ArgumentNullException">StringEncoding must not be null</exception>
        public static Encoding StringEncoding
        {
            get => stringEncoding;
            set => stringEncoding = value ?? throw new ArgumentNullException(nameof(StringEncoding));
        }
```
Can't be `null` now* as the setter validates that.
Otherwise LGTM.

\* except with some unsafe hacks

Reply:
Yes, you are right. I am used to lazy initialization but this time is useless. I will change this right now. Thanks again 👍